### PR TITLE
Changing Storage API key from URI to String

### PR DESCRIPTION
**Pull request by suvodeep-pyne**
Motivation: String is more generic and gives flexibility to the storage implementation to structure their own keys. 
For example, the storage layer may use a JSON string as an identifier. This is not currently a use case, but going forward it would probably be better to not impose an unnecessary restriction on a top level API.

Tested separately on dev clusters.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java [hunk 1]
```diff
@@ -426,7 +426,7 @@ private void addProjectToProjectVersions(Connection connection,
       int version,
       File localFile,
       String uploader,
-      String uri) throws ProjectManagerException {
+      String storage_key) throws ProjectManagerException {
```
Comment (ameyamk) [inline azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java]: storageLocation perhaps?

Comment (ameyamk) [inline azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java]: other idea is storageLocatorKey ?

Comment (suvodeep-pyne) [inline azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java]: updated to `resourceId`

**Comment by suvodeep-pyne** [comment]
Thanks @chengren311 . updated PR description.

**Pull request merged**